include/OsmAndCore/ITypefaceFinder.h

```diff
@@ -18,18 +21,29 @@
 
 namespace OsmAnd
 {
+    using TFontPtr = std::unique_ptr<hb_font_t, std::function<void (hb_font_t*)>>;
     class OSMAND_CORE_API ITypefaceFinder
     {
         Q_DISABLE_COPY_AND_MOVE(ITypefaceFinder);
 
     public:
         struct OSMAND_CORE_API Typeface Q_DECL_FINAL
         {
-            Typeface(const sk_sp<SkTypeface>& skTypeface, hb_face_t* hbTypeface);
+            Typeface(const sk_sp<SkTypeface>& skTypeface,
+                     TFontPtr hbFont_,
+                     std::set<uint32_t> delCodePoints,
+                     uint32_t repCodePoint);
             ~Typeface();
 
             sk_sp<SkTypeface> skTypeface;
-            std::shared_ptr<hb_face_t> hbTypeface;
+
+            TFontPtr hbFont;
```

Comment:
You don't need TFontPtr with that signature here I guess

Reply:
What do you mean?

Reply:
Double-checked - indeed, for unique_ptr that's needed (to specify deleter type in template). 

Btw, why do you need unique_ptr there on contrary to shared_ptr?

Reply:
Because there are no reason to use `shared_ptr` (there is no need to share this resource, at least for now).

Reply:
It's about [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

Reply:
See
https://en.cppreference.com/w/cpp/memory/unique_ptr/get
vs 
https://en.cppreference.com/w/cpp/memory/shared_ptr/get

When Typeface is passed as const, unqiue_ptr does not propagate const to it's content - thus making modification possible, which is not great (see comments below)

Reply:
Implemented using `shared_ptr`